adapter/test/stdio/stdio_adapter_mode_test.cpp

```diff
@@ -0,0 +1,351 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Distributed under BSD 3-Clause license.                                   *
+ * Copyright by The HDF Group.                                               *
+ * Copyright by the Illinois Institute of Technology.                        *
+ * All rights reserved.                                                      *
+ *                                                                           *
+ * This file is part of Hermes. The full Hermes copyright notice, including  *
+ * terms governing use, modification, and redistribution, is contained in    *
+ * the COPYING file, which can be found at the top directory. If you do not  *
+ * have access to the file, you may request a copy from help@hdfgroup.org.   *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#include <catch_config.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include <experimental/filesystem>
+#include <iostream>
+
+#if HERMES_INTERCEPT == 1
+#include <hermes/adapter/stdio.h>
+#endif
+
+namespace fs = std::experimental::filesystem;
+
+namespace hermes::adapter::stdio::test {
+struct Arguments {
+  std::string filename = "test.dat";
+  std::string directory = "/tmp";
+  size_t request_size = 65536;
+};
+struct Info {
+  int rank = 0;
+  int comm_size = 1;
+  std::string write_data;
+  std::string read_data;
+  std::string new_file;
+  std::string existing_file;
+  std::string new_file_cmp;
+  std::string existing_file_cmp;
+  size_t num_iterations = 64;
+  unsigned int offset_seed = 1;
+  unsigned int rs_seed = 1;
+  unsigned int temporal_interval_seed = 5;
+  size_t total_size;
+  size_t stride_size = 512;
+  unsigned int temporal_interval_ms = 1;
+  size_t small_min = 1, small_max = 4 * 1024;
+  size_t medium_min = 4 * 1024 + 1, medium_max = 256 * 1024;
+  size_t large_min = 256 * 1024 + 1, large_max = 3 * 1024 * 1024;
+};
+}  // namespace hermes::adapter::stdio::test
+hermes::adapter::stdio::test::Arguments args;
+hermes::adapter::stdio::test::Info info;
+std::string gen_random(const int len) {
+  std::string tmp_s;
+  static const char alphanum[] =
+      "0123456789"
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+      "abcdefghijklmnopqrstuvwxyz";
+
+  srand(100);
+
+  tmp_s.reserve(len);
+
+  for (int i = 0; i < len; ++i)
+    tmp_s += alphanum[rand() % (sizeof(alphanum) - 1)];
+
+  return tmp_s;
+}
+int init(int* argc, char*** argv) {
+  fs::path fullpath = args.directory;
+  fullpath /= args.filename;
+  info.new_file = fullpath.string() + "_new";
+  info.existing_file = fullpath.string() + "_ext";
+  info.new_file_cmp = fullpath.string() + "_new_cmp";
+  info.existing_file_cmp = fullpath.string() + "_ext_cmp";
```

Comment:
Do these filenames need the PID to avoid conflicts on ares?

Reply:
Yes.